kyo-offheap/native/src/main/java/lang/foreign/Arena.scala

```diff
@@ -0,0 +1,42 @@
+package java.lang.foreign
+
+import scala.scalanative.unsafe._
+import scala.scalanative.libc.stdlib.{malloc, free}
+import java.util.concurrent.ConcurrentLinkedQueue
+
+/** A minimal stub for Java's Arena using Scala Native.
+  *
+  * An Arena is used to track off-heap allocations so that they can be
+  * automatically deallocated when the scope ends.
+  */
+final class Arena extends AutoCloseable {
+  private val allocations = new ConcurrentLinkedQueue[MemorySegment]()
```

Comment:
One more idea. Maybe instead of storing whole `MemorySegment`s, we can story a mere `Ptr`s here?
We need only the Ptr because we need to `free` it. We don't need to know the `byteSize`.